libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxSpeechApi.kt

```diff
@@ -73,12 +118,50 @@ class MapboxSpeechApi @JvmOverloads constructor(
      */
     fun clean(announcement: SpeechAnnouncement) {
         voiceAPI.clean(announcement)
+        VoiceInstructionsParser.parse(announcement).onValue {
+            synchronized(cachedFilesLock) {
+                val value = cachedFiles[it]
+                // when we clear fallback announcement, there is a chance we will remove the key
+                // from map and not remove the file itself
+                // since for fallback SpeechAnnouncement file is null
+                if (value?.announcement == announcement) {
+                    cachedFiles.remove(it)
+                }
+            }
+        }
+    }
+
+    internal fun predownload(instructions: List<VoiceInstructions>) {
+        mainJobController.scope.launch {
+            instructions.forEach { instruction ->
+                val typeAndAnnouncement = VoiceInstructionsParser.parse(instruction).value
+                if (typeAndAnnouncement != null && !hasTypeAndAnnouncement(typeAndAnnouncement)) {
+                    predownloadScope.launch {
```

Comment:
why do you start processing from main scope and continue it in predownload? maybe it makes sense to use only predownload?